### PR TITLE
sdk v50 proposal compatibility

### PR DESCRIPTION
**Pull request by Reecepbcups**
(used in x/POA module)

SDK v50 added more governance params in the proposal.json.

The new functions are suffixed with the IBC Go / interchaintest version (ex: BuildProposal and BuildProposalV8). Should we instead prefix with SDK versions?

## after merge todo
- tag v7 and v8 releases

### CODE REVIEW

**Comment by Reecepbcups** [comment]
Ahhh we need to be backwards comptable for SDK v47 (v7) ->v50 (v8) upgrades

So multiple structs will be required. Suffix based off the IBC go version instead of SDK (Getv8Proposal)

File: chain/cosmos/cosmos_chain.go [hunk 1]
```diff
@@ -439,7 +444,9 @@ func (c *CosmosChain) SubmitProposal(ctx context.Context, keyName string, prop T
 }
 
 // Build a gov v1 proposal type.
-func (c *CosmosChain) BuildProposal(messages []cosmosproto.Message, title, summary, metadata, depositStr string) (TxProposalv1, error) {
+//
+// The proposer field should only be set for IBC-Go v8 / SDK v50 chains.
```
Comment (boojamya) [inline chain/cosmos/cosmos_chain.go:448]: We should also add some sort of note that the `expedited` field is also not ingested for non-sdk50 chains.

**Review by boojamya — APPROVED** [review]
Nice, looks good!
One nit. Approving to not block.

File: chain/cosmos/cosmos_chain.go (hunk 1 above)
Comment (Reecepbcups) [reply chain/cosmos/cosmos_chain.go:448]: this will not matter so long as depositStr is blank. Will error out if they try anything for legacy chains

Comment (boojamya) [reply chain/cosmos/cosmos_chain.go:448]: Wait how does depositStr come into play here?

Comment (Reecepbcups) [reply chain/cosmos/cosmos_chain.go:448]: oops! sorry I mis-copied `proposer`

Comment (boojamya) [reply chain/cosmos/cosmos_chain.go:448]: Copy that. It was more of a guidance thing for users. But yea, it's self explanatory, should be fine.

**Pull request merged**